### PR TITLE
[gosrc2cpg] - Added third party rules

### PR DESCRIPTION
**Pull request by ankit-privado**

### CODE REVIEW

File: rules/sinks/third_parties/sdk/aftership/go.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+#  The id follows a format : "ThirdParties.SDK.<THIRD_PARTY_ORGANISATION>.<SUB_ORGANISATION_IF_APPLICABLE>"
+
+sinks:
+
+  - id: ThirdParties.SDK.Aftership
+    name: Aftership
+    domains:
+      - "aftership.com"
+    patterns:
+      - "(?i)(github.com)[\\/](jackharley7|aftership)[\\/](aftership-sdk-go|).*"
```
Comment (hiteshbedre) [inline rules/sinks/third_parties/sdk/aftership/go.yaml:12]: Remove `|` at the end.

Comment (ankit-privado) [reply rules/sinks/third_parties/sdk/aftership/go.yaml]: yes done.

File: rules/sinks/third_parties/sdk/auth0/go.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+#  The id follows a format : "ThirdParties.SDK.<THIRD_PARTY_ORGANISATION>.<SUB_ORGANISATION_IF_APPLICABLE>"
+
+sinks:
+
+  - id: ThirdParties.SDK.Auth0
+    name: Auth0
+    domains:
+      - "auth0.com"
+    patterns:
+      - "(?i)(github.com)(/)(auth0-community|pulumi|auth0-lab)(/)(go-auth0|pulumi-auth0\\/sdk\\/v2\\/go\\/auth0|fga-go-sdk).*"
```
Comment (hiteshbedre) [inline rules/sinks/third_parties/sdk/auth0/go.yaml:12]: Remove escape characters.

File: rules/sinks/third_parties/sdk/callr/go.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+#  The id follows a format : "ThirdParties.SDK.<THIRD_PARTY_ORGANISATION>.<SUB_ORGANISATION_IF_APPLICABLE>"
+
+sinks:
+
+  - id: ThirdParties.SDK.Callr
+    name: Callr
+    domains:
+      - "callr.com"
+    patterns:
+      - "(?i)(github.com)(/)(THECALLR)(/)(sdk-go).*"
```
Comment (hiteshbedre) [inline rules/sinks/third_parties/sdk/callr/go.yaml:12]: make everything in small case.

File: rules/sinks/third_parties/sdk/auth0/go.yaml (hunk 2 above)
Comment (ankit-privado) [reply rules/sinks/third_parties/sdk/auth0/go.yaml]: yes changed now.

File: rules/sinks/third_parties/sdk/checkout/go.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+#  The id follows a format : "ThirdParties.SDK.<THIRD_PARTY_ORGANISATION>.<SUB_ORGANISATION_IF_APPLICABLE>"
+
+sinks:
+
+  - id: ThirdParties.SDK.Checkout
+    name: Checkout
+    domains:
+      - "checkout.com"
+    patterns:
+      - "(?i)(github.com)(/)(checkout)(/)(checkout-sdk-go\\/payments).*"
```
Comment (hiteshbedre) [inline rules/sinks/third_parties/sdk/checkout/go.yaml]: remove escape character.

File: rules/sinks/third_parties/sdk/consul/go.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+#  The id follows a format : "ThirdParties.SDK.<THIRD_PARTY_ORGANISATION>.<SUB_ORGANISATION_IF_APPLICABLE>"
+
+sinks:
+
+  - id: ThirdParties.SDK.Consul
+    name: Consul
+    domains:
+      - "consul.io"
+    patterns:
+      - "(?i)(github.com|gopkg.in)(/)(launchdarkly|yingyingtang-brex|hashicorp)(/)(go-server-sdk.v4|go-server-sdk-consul|go-server-sdk\\/ldconsul|go-client.v4\\/ldconsul|go-client\\/ldconsul|packer-plugin-sdk\\/template|waypoint-plugin-sdk).*"
```
Comment (hiteshbedre) [inline rules/sinks/third_parties/sdk/consul/go.yaml]: remove escape character

File: rules/sinks/third_parties/sdk/callr/go.yaml (hunk 3 above)
Comment (ankit-privado) [reply rules/sinks/third_parties/sdk/callr/go.yaml:12]: repo name itself was capital letter.

File: rules/sinks/third_parties/sdk/consul/go.yaml (hunk 5 above)
Comment (hiteshbedre) [reply rules/sinks/third_parties/sdk/consul/go.yaml]: remove `.v4` as it not used directly under import statement.

Comment (ankit-privado) [reply rules/sinks/third_parties/sdk/consul/go.yaml]: okay okay.

Comment (ankit-privado) [reply rules/sinks/third_parties/sdk/consul/go.yaml]: yes removed already.

File: rules/sinks/third_parties/sdk/hubspot/go.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+#  The id follows a format : "ThirdParties.SDK.<THIRD_PARTY_ORGANISATION>.<SUB_ORGANISATION_IF_APPLICABLE>"
+
+sinks:
+
+  - id: ThirdParties.SDK.Hubspot
+    name: Hubspot
+    domains:
+      - "hubspot.com"
+    patterns:
+      - "(?i)(github.com)(/)(getwagz|Getwagz)(/)(hubspot-sdk).*"
```
Comment (hiteshbedre) [inline rules/sinks/third_parties/sdk/hubspot/go.yaml:12]: convert `getwagz|Getwagz` to `getwagz`

File: rules/sinks/third_parties/sdk/knadh/go.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+
+#  Sink rule for ThirdParty SDK
+#  The id follows a format : "ThirdParties.SDK.<THIRD_PARTY_ORGANISATION>.<SUB_ORGANISATION_IF_APPLICABLE>"
+
+sinks:
```
Comment (hiteshbedre) [inline rules/sinks/third_parties/sdk/knadh/go.yaml:5]: Verify if this library deals with PII @ankit-privado

Comment (ankit-privado) [reply rules/sinks/third_parties/sdk/knadh/go.yaml:5]: Here is one use case I found https://github.com/knadh/koanf#reading-from-an-s3-bucket.

Comment (ankit-privado) [reply rules/sinks/third_parties/sdk/knadh/go.yaml:5]: @hiteshbedre

File: rules/sinks/third_parties/sdk/hubspot/go.yaml (hunk 6 above)
Comment (ankit-privado) [reply rules/sinks/third_parties/sdk/hubspot/go.yaml:12]: done.

File: rules/sinks/third_parties/sdk/microsoft/go.yaml [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+#  The id follows a format : "ThirdParties.SDK.<THIRD_PARTY_ORGANISATION>.<SUB_ORGANISATION_IF_APPLICABLE>"
+
+sinks:
+
+  - id: ThirdParties.SDK.Microsoft
+    name: Microsoft
+    domains:
+      - "azure.microsoft.com"
+    patterns:
+      - "(?i)(github.com)(/)(Azure|AzureAD|microsoft|MSOpenTech|kris-nova|kubicorn|pulumi|vmware|SsrCoder|pmalek-sumo|wiz-sec|jackesdavid|msopentech|discentem|dennis-menge|veertuinc|beyondnetworks|peterzeller|dbaur|larstore|johnstairs|tadelesh|hashicorp|tombuildsstuff)(/)(azure-pipeline-go|azure-sdk-for-go|kiota/abstractions/go|kubicorn/cloud/azure|pulumi-azure|vra-sdk-go/pkg/client/fabric_azure_storage_account|onedrive-sdk-golang|go-autorest|go-ntlmssp|vault-plugin-auth-azure|go-azure-helpers|giovanni|azure-devops-go-api|azure-service-bus-go|go-amqp|azure-amqp-common-go|microsoft-authentication-library-for-go).*"
```
Comment (hiteshbedre) [inline rules/sinks/third_parties/sdk/microsoft/go.yaml:12]: Reverify dependencies belong to Microsoft only.

**Review by hiteshbedre — APPROVED** [review]
LGTM

File: rules/sinks/third_parties/sdk/knadh/go.yaml (hunk 7 above)
Comment (ankit-privado) [reply rules/sinks/third_parties/sdk/knadh/go.yaml:5]: resolved.

**Pull request merged**